CRM/Core/Resources.php

```diff
@@ -881,4 +882,27 @@ private function resolveFileName(&$fileName, $extName) {
     }
   }
 
+  /**
+   * @param string $url
+   * @return string
+   */
+  public function addCacheCode($url) {
+    parse_str(parse_url($url, PHP_URL_QUERY), $queryParts);
+    $existing = isset($queryParts['r']) ? $queryParts['r'] : NULL;
+    $latest = $this->cacheCode;
+
+    if ($existing) {
+      if ($existing === $latest) {
+        return $url; // no need to update
+      }
+      else {
+        return str_replace('r=' . $existing, 'r=' . $latest, $url);
```

Comment:
@totten it's certainly a lot simpler with that change!

I did a quick comparison on the performance of `parse_url` vs. `str_pos` which I mentioned on [Stack Overflow](http://stackoverflow.com/a/42805590/1196369), also in the hope that if I'm wrong about the check someone will point it out.